library/src/scala/tasty/Reflection.scala

```diff
@@ -85,7 +84,6 @@ trait Reflection extends reflect.Types { reflectSelf: CompilerInterface =>
 
     end extension
 
-
     /** Convert this tree to an `quoted.Expr[T]` if the tree is a valid expression or throws */
     extension [T](tree: Tree)
       def asExprOf(using scala.quoted.Type[T])(using QuoteContext): scala.quoted.Expr[T] =
```

Comment:
Can we merge with the previous extension blockby moving `T` to `def asExprOf[T]`?

Reply:
We currently have the limitation that type parameters must be on the extension. 
We cannot write
```
extension (tree: Tree)
      def asExprOf[T] ...
```
to avoid having a `T` that may depend on `tree`.